tracer/test/Datadog.Trace.Tests/Configuration/TracerSettingsTests.cs

```diff
@@ -452,6 +453,52 @@ public void CustomSamplingRules(string value, string expected)
             settings.CustomSamplingRules.Should().Be(expected);
         }
 
+        [Theory]
+        [InlineData("glob", SamplingRulesFormat.Glob)]     // exact match
+        [InlineData("Glob", SamplingRulesFormat.Glob)]     // case-insensitive
+        [InlineData(" glob ", SamplingRulesFormat.Glob)]   // trim whitespace
+        [InlineData("regex", SamplingRulesFormat.Regex)]   // exact match
+        [InlineData("RegEx", SamplingRulesFormat.Regex)]   // case-insensitive
+        [InlineData(" regex ", SamplingRulesFormat.Regex)] // trim whitespace
+        [InlineData("none", SamplingRulesFormat.Unknown)]  // invalid
+        [InlineData("1", SamplingRulesFormat.Unknown)]     // invalid
+        [InlineData(null, SamplingRulesFormat.Regex)]      // null or empty or whitespace defaults to regex
+        [InlineData("", SamplingRulesFormat.Regex)]        // null or empty or whitespace defaults to regex
+        [InlineData("  ", SamplingRulesFormat.Regex)]      // null or empty or whitespace defaults to regex
+        public void CustomSamplingRulesFormat(string value, string expected)
+        {
+            var source = CreateConfigurationSource((ConfigurationKeys.CustomSamplingRulesFormat, value));
+            var telemetry = new ConfigurationTelemetry();
+            var settings = new TracerSettings(source, telemetry);
+
+            // verify setting
+            settings.CustomSamplingRulesFormat.Should().Be(expected);
+
+            // verify telemetry
+            var entries = telemetry.GetQueueForTesting().Where(e => e is { Key: ConfigurationKeys.CustomSamplingRulesFormat });
```

Comment:
TODO: verify entry _count_ (it's usually 1, but 2 if there's an invalid value)

Reply:
The really important thing to verify is that the entry with the _highest_ `seq` has the _actual_ value we're using

Reply:
LGTM!

Reply:
I'm seeing the opposite of what you expect. I think the manual telemetry record is happening _before_ the one we get automatically from the setting system.